src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -75,6 +76,29 @@ internal sealed class DashboardServiceHost : IHostedService
             // Configuration
             builder.Services.AddSingleton(configuration);
 
+            var resourceServiceConfigSection = configuration.GetSection("AppHost:ResourceService");
+            builder.Services.AddOptions<ResourceServiceOptions>()
+                .Bind(resourceServiceConfigSection)
+                .ValidateOnStart();
+            builder.Services.AddSingleton<IValidateOptions<ResourceServiceOptions>, ValidateResourceServiceOptions>();
+
+            // Configure authentication scheme for the dashboard service
+            builder.Services
+                .AddAuthentication()
+                .AddScheme<ResourceServiceApiKeyAuthenticationOptions, ResourceServiceApiKeyAuthenticationHandler>(
+                    ResourceServiceAuthenticationDefaults.AuthenticationScheme,
+                    options => { });
+
+            // Configure authorization policy for the dashboard service
+            builder.Services
+                .AddAuthorizationBuilder()
+                .AddPolicy(
+                    name: ResourceServiceAuthorization.PolicyName,
+                    policy: new AuthorizationPolicyBuilder(
+                        ResourceServiceAuthenticationDefaults.AuthenticationScheme)
+                        .RequireAssertion(_ => true)
```

Comment:
Doesn't this allow everyone through?

In the dashboard I made auth policies I added require a specific claim for the auth method configured, and then added that claim when authenticating with different types of auth.

The benefit of doing this is if there are multiple ways of doing auth configured (e.g. certs and API key) then someone can't pass authz when authenticated with a valid certificate when actually they needed an API key.

Reply:
It allows anyone who authenticates. Authentication is currently the only requirement. However, we need an authorization policy in order to map to the authentication scheme. The auth scheme that's used is determined by the app host's configuration. I'll add a comment to clarify this.

Reply:
Doesn't `RequireAssertion(_ => true)` let everyone through? Shouldn't it have `RequireAuthenticatedUser()`?

Reply:
I tested with the client providing an incorrect API key and it wasn't denied. Can you please double check?

I'm really nervious about the auth that is being added with unit and integration tests.

Reply:
RequireAuthenticatedUser is what it should have

Reply:
That's embarrassing. Somehow this slipped through my manual testing. Updated and tested. Integration tests coming soon.